File: DailyPractice/Week28/chopsticks.cpp
#include<bits/stdc++.h>
#define ll             long long int
#define F              first
#define S              second
#define pb             push_back
#define vi             vector <int>
#define pii            pair <int, int>
using namespace std;
class Solution{
    public:
    void solve(){
        int n, d;
        cin >> n >> d;
        vector<int> v(n);
        for (int i = 0; i < n; i ++){
            cin >> v[i];
        }
        sort(v.begin(), end(v));
        int count = 0;
        for (int i = 0; i < n - 1; i++){
            if(abs(v[i] - v[i + 1]) <= d)
                count++ , i++;
            else
                continue;
        }
        cout << count;
    }
};


int main(){

#ifndef ONLINE_JUDGE
    freopen("input.txt","r",stdin);
    freopen("output.txt","w",stdout);
#endif
    Solution s;
    int t;
    // cin >> t;
    // while(t--){
        s.solve();
        cout<<'\n';
    // }
}

File: DailyPractice/Week28/chopsticks.py
from os import path
import sys 

class Solution:
    def solve(self):
        n , d = map(int, input().split())
        ls = []
        for i in range(n):
            ls.append(int(input()))
        ls.sort()
        i = 0
        count = 0
        while i < len(ls) - 1:
            if abs(ls[i] - ls[i + 1]) <= d:
                count += 1
                i += 2
            else:
                i += 1

        print(count)
        

s = Solution()
if path.exists('input.txt'):
    sys.stdin = open('input.txt', "r")
    sys.stdout = open('output.txt' , "w")
# t = int(input())
# while t != 0:
s.solve()